src/shared/components/oncoprint/OncoprintUtils.ts

```diff
@@ -972,13 +991,54 @@ export function makeClinicalTracksMobxPromise(
                         dataAndColors.categoryToColor,
                         hexToRGBA
                     );
+                    // if default for attribute doesn't already exist
+                    if (
+                        !oncoprint.props.store.defaultClinicalTracksColors[
+                            attribute.displayName
+                        ]
+                    ) {
+                        oncoprint.props.store.setDefaultClinicalTrackColors(
```

Comment:
we should never set state from inside of an invoke (this is known as a side effect)